DigitalLearningSolutions.Data/Services/CustomPromptsService.cs

```diff
@@ -26,21 +31,138 @@ public CustomPromptsService(ICustomPromptsDataService customPromptsDataService)
 
             var result = customPromptsDataService.GetCentreCustomPromptsByCentreId(centreId.Value);
 
-            return new CentreCustomPrompts
+            return new CentreCustomPrompts(result.CentreId, PopulateCustomPromptListFromCentreCustomPromptsResult(result));
+        }
+
+        public CentreCustomPromptsWithAnswers? GetCentreCustomPromptsWithAnswersByCentreIdAndDelegateUser(int? centreId,
```

Comment:
I wonder if we want to ever handle the optional cases - I'm wondering whether accepting int centreId and DelegateUser would be better as then we wouldn't have to return/handle nullables everywhere. If the centreId doesn't exist on the claims, then it's ok to redirect the user to unexpected error page. If we want to be extra sure, we could always check for non-null centre id in the centre admin claim policy..